src/Microsoft.Net.Http.Headers/EntityTagHeaderValue.cs

```diff
@@ -112,6 +112,24 @@ public override int GetHashCode()
             return _tag.GetHashCode() ^ _isWeak.GetHashCode();
         }
 
+        public static bool CompareStrongly(EntityTagHeaderValue left, EntityTagHeaderValue right)
+        {
+            if (left == null || right == null)
+            {
+                return false;
+            }
+            return !left.IsWeak && !right.IsWeak && string.Equals(left.Tag, right.Tag);
+        }
+
+        public static bool CompareWeakly(EntityTagHeaderValue left, EntityTagHeaderValue right)
+        {
+            if (left == null || right == null)
```

Comment:
What should we do about comparing two null objects? I'm currently leaning towards considering that situation as not equal. We could consider returning false for strong comparisons and true for weak comparisons.